conda/core/solve.py

```diff
@@ -520,54 +522,53 @@ def empty_package_list(pkg):
 
     @time_recorder(module_name=__name__)
     def _collect_all_metadata(self, ssc):
-        # add in historically-requested specs
-        ssc.specs_map.update(ssc.specs_from_history_map)
-
-        # these are things that we want to keep even if they're not explicitly specified.  This
-        #     is to compensate for older installers not recording these appropriately for them
-        #     to be preserved.
-        for pkg_name in (
-            "anaconda",
-            "conda",
-            "conda-build",
-            "python.app",
-            "console_shortcut",
-            "powershell_shortcut",
-        ):
-            if pkg_name not in ssc.specs_map and ssc.prefix_data.get(pkg_name, None):
-                ssc.specs_map[pkg_name] = MatchSpec(pkg_name)
-
-        # Add virtual packages so they are taken into account by the solver
-        virtual_pkg_index = {}
-        _supplement_index_with_system(virtual_pkg_index)
-        virtual_pkgs = [p.name for p in virtual_pkg_index.keys()]
-        for virtual_pkgs_name in virtual_pkgs:
-            if virtual_pkgs_name not in ssc.specs_map:
-                ssc.specs_map[virtual_pkgs_name] = MatchSpec(virtual_pkgs_name)
-
-        for prec in ssc.prefix_data.iter_records():
-            # first check: add everything if we have no history to work with.
-            #    This happens with "update --all", for example.
-            #
-            # second check: add in aggressively updated packages
-            #
-            # third check: add in foreign stuff (e.g. from pip) into the specs
-            #    map. We add it so that it can be left alone more. This is a
-            #    declaration that it is manually installed, much like the
-            #    history map. It may still be replaced if it is in conflict,
-            #    but it is not just an indirect dep that can be pruned.
-            if (
-                not ssc.specs_from_history_map
-                or MatchSpec(prec.name) in context.aggressive_update_packages
-                or prec.subdir == "pypi"
-            ):
-                ssc.specs_map.update({prec.name: MatchSpec(prec.name)})
+        if ssc.prune:
+            # When pruning DO NOT consider history of already installed packages when solving.
+            prepared_specs = set(concatv(
+                self.specs_to_remove,
+                self.specs_to_add,
+            ))
```

Comment:
We've also removed usage of `concatv()` (please see https://github.com/conda/conda/pull/12020)

```suggestion
            prepared_specs = set(*self.specs_to_remove, *self.specs_to_add)
```